types/node/test/http2.ts

```diff
@@ -67,7 +67,7 @@ import { URL } from "node:url";
     http2Session.on("connect", (session: Http2Session, socket: Socket) => {});
     http2Session.on("error", (err: Error) => {});
     http2Session.on("frameError", (frameType: number, errorCode: number, streamID: number) => {});
-    http2Session.on("goaway", (errorCode: number, lastStreamID: number, opaqueData: Buffer) => {});
```

Comment:
🤔 I expected this line to fail compilation, but the old signature is actually still accepted because it matches a different overload with a "catch all" handler definition ☝️ 
```
on(event: string | symbol, listener: (...args: any[]) => void): this;
```

I wonder why we have this "catch all" overload, maybe to account for future usage? But unfortunately it relaxes validation for `'goaway'` and might prevent uncovering incompatible usage.

Any suggestion from the maintainers?